### PR TITLE
Osc/mt v4 tmp

### PR DESCRIPTION
**Pull request by xinzhao3**

### CODE REVIEW

File: opal/mca/common/ucx/common_ucx_wpool.c [hunk 1]
```diff
@@ -1228,22 +1228,25 @@ opal_common_ucx_tlocal_fetch_spath(opal_common_ucx_wpmem_t *mem, int target)
 }
 
 OPAL_DECLSPEC int
-opal_common_ucx_flush(ucp_ep_h ep, ucp_worker_h worker,
+opal_common_ucx_flush(opal_common_ucx_winfo_t *winfo, int target,
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.c]: change the name to `opal_common_ucx_winfo_flush` to indicate that it is specific for wpool.

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 2]
```diff
@@ -246,11 +246,58 @@ OPAL_DECLSPEC int opal_common_ucx_wpmem_flush(opal_common_ucx_wpmem_t *mem,
...
+                                           msg ? msg : __func__,
+                                           UCS_PTR_STATUS(request),
+                                           ucs_status_string(UCS_PTR_STATUS(request)));
+                   return OPAL_ERROR;
+                }
+                break;
+            }
+            ctr--;
+        } while (ctr > 0 && ret > 0);
+        opal_mutex_unlock(&winfo->mutex);
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h:305]: call opal_progress()

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 3]
```diff
@@ -246,11 +246,58 @@ OPAL_DECLSPEC int opal_common_ucx_wpmem_flush(opal_common_ucx_wpmem_t *mem,
...
+    do {
+        ctr = opal_common_ucx.progress_iterations;
+        opal_mutex_lock(&winfo->mutex);
+        do {
+            ret = ucp_worker_progress(winfo->worker);
+            status = opal_common_ucx_request_status(request);
+            if (status != UCS_INPROGRESS) {
+                ucp_request_free(request);
+                if (OPAL_LIKELY(UCS_OK == status)) {
+                    return OPAL_SUCCESS;
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h]: need to unlock

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 4]
```diff
@@ -246,11 +246,58 @@ OPAL_DECLSPEC int opal_common_ucx_wpmem_flush(opal_common_ucx_wpmem_t *mem,
...
+                } else {
+                    MCA_COMMON_UCX_VERBOSE(1, "%s failed: %d, %s",
+                                           msg ? msg : __func__,
+                                           UCS_PTR_STATUS(request),
+                                           ucs_status_string(UCS_PTR_STATUS(request)));
+                   return OPAL_ERROR;
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h:299]: unlock

File: opal/mca/common/ucx/common_ucx_wpool.h [hunk 5]
```diff
@@ -246,11 +246,58 @@ OPAL_DECLSPEC int opal_common_ucx_wpmem_flush(opal_common_ucx_wpmem_t *mem,
...
+                }
+                break;
+            }
+            ctr--;
+        } while (ctr > 0 && ret > 0);
```
Comment (artpol84) [inline opal/mca/common/ucx/common_ucx_wpool.h]: status == UCS_INPROGRESS

**Review by artpol84 — CHANGES_REQUESTED** [review]

**Comment by xinzhao3** [comment]
@artpol84 could you review again?

**Pull request merged**